app/src/main/kotlin/com/wire/android/notification/CallNotificationManager.kt

```diff
@@ -0,0 +1,138 @@
+                .Builder(CHANNEL_ID, NotificationManagerCompat.IMPORTANCE_MAX)
+                .setName(CHANNEL_NAME)
+                .build()
+
+            notificationManager.createNotificationChannel(notificationChannel)
+        }
+    }
+
+    private fun getNotification(call: Call, userId: QualifiedID): Notification {
+        val conversationIdString = call.conversationId.toString()
+        val userIdString = userId.toString()
+        val title = getNotificationTitle(call)
+        val content = getNotificationBody(call)
+
+        return getCompatNotification(title, content, conversationIdString, userIdString)
+    }
+
+    private fun getNotificationBody(call: Call) =
+        when (call.conversationType) {
+            Conversation.Type.GROUP -> {
+                val name = call.callerName ?: context.getString(R.string.notification_call_default_caller_name)
+                (call.callerTeamName?.let { "$name @$it" } ?: name)
+                    .let { context.getString(R.string.notification_group_call_content, it) }
+            }
+            else -> context.getString(R.string.notification_call_content)
+        }
+
+    private fun getNotificationTitle(call: Call): String =
+        when (call.conversationType) {
+            Conversation.Type.GROUP -> call.conversationName ?: context.getString(R.string.notification_call_default_group_name)
+            else -> {
+                val name = call.callerName ?: context.getString(R.string.notification_call_default_caller_name)
+                call.callerTeamName?.let { "$name @$it" } ?: name
+            }
+        }
+
+    private fun getCompatNotification(
+        title: String,
+        content: String,
+        conversationIdString: String,
+        userIdString: String
+    ) = NotificationCompat.Builder(context, CHANNEL_ID)
+        .setContentTitle(title)
+        .setContentText(content)
+        .addAction(getDeclineCallCompatAction(conversationIdString, userIdString))
+        .addAction(getOpenCallCompatAction(conversationIdString))
+        .setOngoing(true)
+        .setPriority(NotificationCompat.PRIORITY_MAX)
+        .setSmallIcon(R.drawable.notification_icon_small)
+        .setContentIntent(fullScreenCallPendingIntent(context, conversationIdString))
+        .setFullScreenIntent(fullScreenCallPendingIntent(context, conversationIdString), true)
+        .setAutoCancel(true)
+        .build()
+
+    private fun getOpenCallAction(conversationId: String) = Notification.Action.Builder(
+        null,
+        context.getString(R.string.notification_action_open_call),
+        openCallPendingIntent(context, conversationId)
+    )
+        .build()
+
+    private fun getDeclineCallAction(conversationId: String, userId: String) = Notification.Action.Builder(
+        null,
+        context.getString(R.string.notification_action_decline_call),
+        declineCallPendingIntent(context, conversationId, userId)
+    )
+        .build()
+
```

Comment:
maybe we can create a function for the builder with 2 arguments (string, pending intent ), 
so you will reduce the boilerplate code